src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1045,6 +1049,28 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 	foreach(rangeTableCell, rangeTableList)
 	{
 		RangeTblEntry *rangeTableEntry = (RangeTblEntry *) lfirst(rangeTableCell);
+		bool referenceTable = false;
+
+		if (IsDistributedTable(rangeTableEntry->relid))
+		{
+			DistTableCacheEntry *distTableEntry =
+				DistributedTableCacheEntry(rangeTableEntry->relid);
+
+			if (distTableEntry->replicationModel == REPLICATION_MODEL_2PC)
```

Comment:
Also, probably looking into `partitionMethod` would make it more readable and prevent any future bugs where we could use `replicationModel2PC` for other purposes etc.